core/src/main/java/org/infinispan/hadoop/impl/InfinispanRecordWriter.java

```diff
@@ -0,0 +1,45 @@
+package org.infinispan.hadoop.impl;
+
+import org.apache.hadoop.mapreduce.RecordWriter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.infinispan.client.hotrod.RemoteCache;
+
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+/**
+ * Write output to the configured cache.
+ *
+ * @param <K> Key type
+ * @param <V> Value type
+ * @author gustavonalle
+ * @since 0.1
+ */
+public class InfinispanRecordWriter<K, V> extends RecordWriter<K, V> {
+   private final RemoteCache<K, V> remoteCache;
+   private final int writeBatchSize;
+   private final Map<K, V> batch;
+
+   public InfinispanRecordWriter(RemoteCache<K, V> remoteCache, int writeBatchSize) {
+      this.remoteCache = remoteCache;
+      this.writeBatchSize = writeBatchSize;
+      this.batch = new LinkedHashMap<>(writeBatchSize);
```

Comment:
just a curiosity, why a ```LinkedHashMap```? keeping the insertion order wouldn't bring any advantage I believe...